src/debugger.ts

```diff
@@ -46,7 +46,7 @@ function getFieldAndExecutable(config: vscode.DebugConfiguration): [keyof vscode
 function changeConfigForSip(config: vscode.DebugConfiguration, executableFieldName: string, executionInfo: MirrordExecution) {
 	if (config.type === "node-terminal") {
 		const command = config[executableFieldName];
-		if (command === null) {
+		if (command == null) {
 			return;
```

Comment:
Isn't this a failure if we reach the `return`?

```suggestion
		if (!command) {
			throw new Exception(`Command not found!`);
```

Reply:
By returning and not throwing an exception, we're saying: if this is invalid, vscode will fail to launch and show the normal error it shows in that case. If this is somehow valid, maybe we're lucky and whatever is going to run does not need SIP. Of course the tradeoff is that maybe it does and then we don't handle SIP and mirrord would fail to load without saying anything. So throwing an Exception is better?

Reply:
Hmm, if it's possible to get lucky here, then maybe add a warning mentioning that this might not work due to SIP.

Also, looking at the whole function, I would suggest a small refactor here.

```ts
function changeConfigForSip(config: vscode.DebugConfiguration, executableFieldName: string, executionInfo: MirrordExecution): String | undefined {
  if (config.type === "node-terminal") {
    const command = config[executableFieldName];
    if (!command) {
      return undefined;
    } else {
      // The command could have single quotes, and we are putting the whole command in single quotes in the changed command.
      // So we replace each `'` with `'\''` (closes the string, concats an escaped single quote, opens the string)
      const escapedCommand = command.replaceAll("'", "'\\''");
      const sh = executionInfo.patchedPath ?? vscode.env.shell;

      const libraryPath = executionInfo.env.get(DYLD_ENV_VAR_NAME);

      // Run the command in a SIP-patched shell, that way everything that runs in the original command will be SIP-patched
      // on runtime.
      return `echo '${escapedCommand}' | ${DYLD_ENV_VAR_NAME}=${libraryPath} ${sh} -is`;
	}
  } else if (executionInfo.patchedPath) {
    return executionInfo.patchedPath!;
  } else {
    // what do we return here?
  }
}
```

And change the call-site to be the one modifying `config[executableFileName]` with the results of this. With the goal here being to drop mutation from helper functions. wdyt?